ducktape/cluster/vagrant.py

```diff
@@ -24,10 +28,53 @@ class VagrantCluster(JsonCluster):
     advertised, this assumes that the Vagrant VM's name is a routeable hostname on all the hosts.
     """
 
-    def __init__(self):
-        hostname, ssh_hostname, username, flags = None, None, None, ""
-        nodes = []
+    def __init__(self, *args, **kwargs):
+        is_read_from_file = False
 
+        cluster_file = kwargs.get("cluster_file")
+        if cluster_file is not None:
+            try:
+                cluster_json = json.load(open(os.path.abspath(cluster_file)))
+                is_read_from_file = True
+            except IOError:
+                # It is OK if file is not found. Call vagrant ssh-info to read the cluster info.
+                pass
```

Comment:
I agree we should print a message to make that explicit.

With regard to your first point, are you saying that I should move `is_read_from_file = False` from its current position to the except block? What would be the value of `is_read_from_file` if cluster_file is None?